### PR TITLE
Test loading

### PR DESCRIPTION
**Pull request by NicoRenaud**
changed the way we handle the loading of pretrained models to make it easier. 
The saved model contains all the parameters relative to the training and can be reused during the creation of the solver.

### CODE REVIEW

File: graphprot/NeuralNet.py [hunk 1]
```diff
@@ -17,51 +17,54 @@
...
                  batch_size=32, percent=[0.8, 0.2], index=None, database_eval=None,
-                 class_weights=None, task='class', classes=[0, 1], threshold=4):
+                 class_weights=None, task='class', classes=[0, 1], threshold=4, pretrained_model=None):
+
+        # load the input data or a pretrained model
+        if pretrained_model is None:
+            for k, v in dict(locals()).items():
+                if k not in ['self', 'database', 'Net', 'database_eval']:
+                    self.__setattr__(k, v)
+
```
Comment (manonreau) [inline graphprot/NeuralNet.py:34]: Very nice and efficient.

File: graphprot/NeuralNet.py [hunk 2]
```diff
@@ -17,51 +17,54 @@
...
+        # PreCluster(dataset, method='mcl')
 
+        # divide the dataset
         train_dataset, valid_dataset = DivideDataSet(
-            dataset, percent=percent)
+            dataset, percent=self.percent)
+
+        # dataloader
+        self.train_loader = DataLoader(
+            train_dataset, batch_size=self.batch_size, shuffle=False)
```
Comment (manonreau) [inline graphprot/NeuralNet.py:51]: Shouldn't we shuffle the data automatically ? or add it as an option ?

File: graphprot/NeuralNet.py [hunk 3]
```diff
@@ -17,51 +17,54 @@
...
         if database_eval is not None:
-            valid_dataset = HDF5DataSet(root='./', database=database_eval, index=index,
-                                        node_feature=node_feature, edge_feature=edge_feature,
-                                        target=target)
+            valid_dataset = HDF5DataSet(root='./', database=database_eval, index=self.index,
+                                        node_feature=self.node_feature, edge_feature=self.edge_feature,
+                                        target=self.target)
+            self.valid_loader = DataLoader(
+                valid_dataset, batch_size=self.batch_size, shuffle=False)
             print('Independent validation set loaded')
```
Comment (manonreau) [inline graphprot/NeuralNet.py:59]: Same here

File: graphprot/NeuralNet.py [hunk 4]
```diff
@@ -154,27 +161,27 @@ def plot_acc(self):
...
+        """
 
         import matplotlib.pyplot as plt
-            
-        try :
+
+        try:
 
             hitrate = self.get_metrics(data, threshold).HitRate()
-
```
Comment (manonreau) [inline graphprot/NeuralNet.py:171]:
```
        hitrate = list(self.get_metrics(data, threshold).HitRate())

        nb_models = len(hitrate)
        X = range(1, nb_models + 1)

        if mode == 'percentage' :
            tot_hit = hitrate.count(1)
            hitrate = [x/tot_hit for x in hitrate]

```

Comment (manonreau) [reply graphprot/NeuralNet.py:171]: The hitrate should be the number of hits at rank X divided by the total number of hits.

**Review by manonreau — APPROVED** [review]

File: graphprot/NeuralNet.py (hunk 2 above)
Comment (NicoRenaud) [reply graphprot/NeuralNet.py:51]: I think we should. I thought you wrote that but it might be me :) I'll add it as an option

File: graphprot/NeuralNet.py (hunk 4 above)
Comment (NicoRenaud) [reply graphprot/NeuralNet.py:171]: yes it should :) I'll fix that

Comment (manonreau) [reply graphprot/NeuralNet.py:171]: My bad, I wrote that !:D

Comment (NicoRenaud) [reply graphprot/NeuralNet.py:171]: plus hitrate is a np array so you can divide it directly by a scalar (or an array of same size), i.e. `hitrate /= hitrate.sum()`

**Review by manonreau — APPROVED** [review]
